src/Draco.Compiler/Internal/Syntax/Syntax.xml

```diff
@@ -401,7 +505,26 @@
     </Field>
   </Node>
 
-  <Node Name="ParameterSyntax" Base="SyntaxNode">
+  <AbstractNode Name="ParameterSyntaxBase" Base="SyntaxNode">
+    <Documentation>
+      A single parameter in a function parameter list.
+    </Documentation>
+  </AbstractNode>
+
+  <Node Name="ThisParameterSyntax" Base="ParameterSyntaxBase">
+    <Documentation>
+      The this parameter at the first place in a parameter list.
+    </Documentation>
+
+    <Field Name="ThisToken" Type="SyntaxToken">
+      <Documentation>
+        The this keyword, representing the type instance.
+      </Documentation>
+      <Token Kind="KeywordThis" />
+    </Field>
+  </Node>
+
+  <Node Name="ParameterSyntax" Base="ParameterSyntaxBase">
```

Comment:
Bug, you mess up syntax ordering of the tree by inheritance, the ellipsis token will be the last child